### PR TITLE
update: adds fulltext to base and jats parsers

### PR DESCRIPTION
**Pull request by seasidesparrow**
modified:   adsingestp/parsers/base.py
 	modified:   adsingestp/parsers/jats.py
 	modified:   tests/stubdata/output/jats_a+a_subtitle.json
 	modified:   tests/stubdata/output/jats_aip_amjph_90_286.json
 	modified:   tests/stubdata/output/jats_aj_158_4_139.json
 	modified:   tests/stubdata/output/jats_apj_859_2_101.json
 	modified:   tests/stubdata/output/jats_iop_ansnn_12_2_025001.json
 	modified:   tests/stubdata/output/jats_iop_apj_923_1_47.json
 	modified:   tests/stubdata/output/jats_iop_no_contribs.json
 	modified:   tests/stubdata/output/jats_iop_no_orcid_tag.json
 	modified:   tests/stubdata/output/jats_iucr_d-60-02355.json
 	modified:   tests/stubdata/output/jats_iucr_d-75-00616.json
 	modified:   tests/stubdata/output/jats_mnras_493_1_141.json
 	modified:   tests/stubdata/output/jats_nature_41467_2023_Article_40261_nlm.json
 	modified:   tests/stubdata/output/jats_phrvd_106_023001.json
 	modified:   tests/stubdata/output/jats_pnas_1715554115.json
 	modified:   tests/stubdata/output/jats_springerEarly_ExA_s10686-023-09907-7.json
 	modified:   tests/stubdata/output/jats_springer_EPJC_s10052-023-11699-1.json
 	modified:   tests/stubdata/output/jats_springer_EPJC_s10052-023-11733-2.json
 	modified:   tests/stubdata/output/jats_springer_NatCo_s41467-023-40272-3.json
 	modified:   tests/stubdata/output/jats_springer_Natur_s41598-023-38673-x.json
 	modified:   tests/stubdata/output/jats_springer_ZaMP_s00033-023-02064-z.json

### CODE REVIEW

File: adsingestp/parsers/jats.py [hunk 1]
```diff
@@ -896,6 +897,11 @@ def _parse_page(self):
                     self.article_meta.find("counts").find("page-count").get("count", "")
                 )
 
+    def _parse_fulltext(self):
+        if self.body is not None:
+            body_text = str(self.body.extract())
+            self.base_metadata["fulltext"] = {"body": body_text, "language": None}
```
Comment (mugdhapolimera) [inline adsingestp/parsers/jats.py:903]: is `language` always going to be set to None for now? Do we have a way of determining the language from the input xml files?

**Review by mugdhapolimera — CHANGES_REQUESTED** [review]
Just a question about the `language` field, but otherwise everything looks good from my end.

File: adsingestp/parsers/jats.py (hunk 1 above)
Comment (seasidesparrow) [reply adsingestp/parsers/jats.py:903]: I would like to keep it as None for now, at least for JATS.  I looked at the documentation for JATS specifically, and did not find attributes available to the `<body>` tag that would indicate the language.  As far as I saw none of our examples uses the `@xml:lang` attribute within the body tag itself.  It is sometimes embedded in sub-elements like `<caption>`.

But I think this prompts the question of whether language as captured at the `<document>` level should be captured within `Document.json`, and I think it should, but not within body if body isn't what's declaring it.

Comment (seasidesparrow) [reply adsingestp/parsers/jats.py:903]: I'll make an issue in ingest_data_model about this.

File: adsingestp/parsers/jats.py [hunk 2]
```diff
@@ -896,6 +897,11 @@ def _parse_page(self):
                     self.article_meta.find("counts").find("page-count").get("count", "")
                 )
 
+    def _parse_fulltext(self):
+        if self.body is not None:
+            body_text = str(self.body.extract())
```
Comment (kelockhart) [inline adsingestp/parsers/jats.py:902]: In the current fulltext pipeline, we do a lot of manipulation on the extracted body (see here for an example: https://github.com/adsabs/ADSfulltext/blob/master/adsft/extraction.py#L710). Will we be passing this through the fulltext pipeline to do that manipulation? What will this parsed fulltext be used for at the moment?

Eventually, I'd like to rethink how the fulltext pipeline operates, in conjunction with these parsers, but that's a bigger task.

File: tests/stubdata/output/jats_a+a_subtitle.json [hunk 3]
```diff
@@ -660,6 +660,9 @@
...
... id=\"InR92\"/><xref ref-type=\"bibr\" rid=\"R92\">Salvaterra et al. 2012</xref>, <xref ref-type=\"bibr\" rid=\"R93\">2013</xref>), which are believed to provide the major contribution to cosmic reionization. It would be very difficult, if not impossible, to obtain the information presented in this work for field galaxies observed even with the best telescopes of the future (JWST, ELT,...). To increase the number of these kind of studies, it is fundamental to develop space satellites with energy range coverage and sensitivity suitable to detect the very high-redshift GRB population (<xref id=\"InR42\"/><xref ref-type=\"bibr\" rid=\"R42\">Ghirlanda et al. 2015</xref>; e.g., THESEUS, Gamow; <xref id=\"InR2\"/><xref ref-type=\"bibr\" rid=\"R2\">Amati et al. 2018</xref>; <xref id=\"InR107\"/><xref ref-type=\"bibr\" rid=\"R107\">Tanvir et al. 2021a</xref>; <xref id=\"InR126\"/><xref ref-type=\"bibr\" rid=\"R126\">White et al. 2021</xref>, respectively). Onboard NIR afterglow detection and preselection of very high-redshift GRBs would be a significant added value to optimize ground-based spectroscopic observations with sensitive medium-high resolution spectrographs.</p>\n</sec>\n</body>"
```
Comment (kelockhart) [inline tests/stubdata/output/jats_a+a_subtitle.json:664]: Are all of these test files safe (non-copyrighted) to have up in a public repo on Github?

**Review by kelockhart — COMMENTED** [review]
I have a couple big-pictures comments/questions, but the code itself looks fine:
* What will the parsed fulltext from this be used for? How will it work in conjunction with the fulltext pipeline?
* Let's confirm that all of the fulltext is safe to put in a public repo

**Comment by seasidesparrow** [comment]
Closing this PR pending additional discussion

**Pull request closed without merging**